internal/resource/url.go

```diff
@@ -319,6 +332,44 @@ func (f *Fetcher) fetchFromDataURL(u url.URL, dest io.Writer, opts FetchOptions)
 	return f.decompressCopyHashAndVerify(dest, bytes.NewBuffer(url.Data), opts)
 }
 
+// FetchFromGCS writes the data stored in a GCS bucket as described by u into dest,returning
+// an error if one is encountered. It looks for the default credentials by querying metadata
+// server on GCE. If it fails to get the credentials, then it will fall back to anonymous
+// credentials to fetch the object content.
+func (f *Fetcher) fetchFromGCS(u url.URL, dest io.Writer, opts FetchOptions) error {
+	ctx := context.Background()
```

Comment:
Can be moved before the `NewClient` call.

Reply:
I can't move this inside `if` condition since it is also used later in the code:  https://github.com/coreos/ignition/pull/1000/commits/c380d56d0da15afb2b7fad55f055b911727e7c97#diff-5a0dbc4ce61f4bd511e01289d0db8fe8R363

Reply:
> Can be moved before the `NewClient` call.


nvm, I found a way to address this comment.

Reply:
Sorry, I missed that the second context wraps the first one.  You were right.